### ISSUE TITLE
Registration email not optional

### ISSUE THREAD

**Issue by Aonodensetsu** [nextcloud/registration#497]
### Steps to reproduce
1. Set up registration with "Email is optional" toggle enabled
2. Try to register without giving an email

### Expected behaviour
The website allows registration to go through

### Actual behaviour
A "this field is required" popup appears

### Server configuration
**Nextcloud version:** 25.0.2

**Comment by nickvergessen** [comment]
That's right, the email is still needed as it's the unique login attribute afterwards. It simply skips the verification where you have to enter a random secret that is emailed to the address.

**Comment by Aonodensetsu** [comment]
(sorry for late reply) Could the username not be treated as unique in that situation? I have a mail server connected to the NC user database which provisions accounts for new users, I'd rather them not use an external email in addition to the provisioned one.

**Comment by vszakats** [comment]
I expected "Email is optional" to mean that email can be left empty on registration (or even better: not asked for at all at this stage), and most importantly: mean that the server doesn't need to collect and store this personal data. If "Email is optional" means "Disable email confirmation", it'd probably be better to call it that.

It's concerning (or at least good to be aware of) that the internal design of Nextcloud server requires an email address for each user profile (if I understand correctly). I'd argue an email address should be optional by design and used for specific purposes based on policies and settings.

**Comment by aradrd** [comment]
A bit late to the party here, but it seems there's another toggle named "Disable email verification" that only shows up when "Email is optional" is disabled. What does that one do then?

**Comment by NeverLoseHope777** [comment]
I would also like to give people who do not have an e-mail address or do not wish to provide one the opportunity to register. Unfortunately, this is currently not possible.

**Comment by smooroodina** [comment]
Why not possible?

 I was able to completely hide the email input process from new users by making a few modifications to this app's source code.

The new registration process I modified is as follows:
- When you press the `Register` button on the login page, a form for entering `loginname` and `password` is displayed immediately without going through the email input process.
- Also, there is no `email` input(pre-filled and disabled) in this form.
![image](https://github.com/user-attachments/assets/7653023a-8fce-4024-934e-ce2d10de536c)
- Fill out the form and submit. New account registration successful.

---

`loginname` is already a unique ID value, so there will be no account conflict issues even if a unique email address is not specified.
since the "Disable email verification" option is already selectable, I see no reason why there shouldn't be an option to completely remove the email requirement from the sign-up process.

I am not yet a contributor to nextcloud. I am willing to submit a pull request after further testing the stability of the modified functionality, but I may have made inappropriate changes to the app's architecture and framework.

So, for now... I just wanted to say that it is fully possible to implement.

**Comment by Areopagitics** [comment]
> Why not possible?
> 
> I was able to completely hide the email input process from new users by making a few modifications to this app's source code.
> 
> The new registration process I modified is as follows:
> 
> * When you press the `Register` button on the login page, a form for entering `loginname` and `password` is displayed immediately without going through the email input process.
> * Also, there is no `email` input(pre-filled and disabled) in this form.
>   ![image](https://private-user-images.githubusercontent.com/81278460/348603351-7653023a-8fce-4024-934e-ce2d10de536c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM2OTUyNzksIm5iZiI6MTcyMzY5NDk3OSwicGF0aCI6Ii84MTI3ODQ2MC8zNDg2MDMzNTEtNzY1MzAyM2EtOGZjZS00MDI0LTkzNGUtY2UyZDEwZGU1MzZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE1VDA0MDkzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjZWQyMzI2YmE5NTNjMzcyMTA2N2NlZDMyNTEzNTM0MDdlNzhlNmI1ODdlNDA4ZDBjN2IzNWQ5MGY5ODE3OTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.YoyOqYOePYxmTPzy52ESnr00BHfRYcZNoRp9mQ1YmYk)
> * Fill out the form and submit. New account registration successful.
> 
> `loginname` is already a unique ID value, so there will be no account conflict issues even if a unique email address is not specified. since the "Disable email verification" option is already selectable, I see no reason why there shouldn't be an option to completely remove the email requirement from the sign-up process.
> 
> I am not yet a contributor to nextcloud. I am willing to submit a pull request after further testing the stability of the modified functionality, but I may have made inappropriate changes to the app's architecture and framework.
> 
> So, for now... I just wanted to say that it is fully possible to implement.

@smooroodina  How is the testing coming along? Any updates on a merge? Thanks!

**Comment by ChrisFa90** [comment]
Hey, I am also wondering if there has been any progress on this Issue. I have the "Email is Optional" Feature enabled, but when I try to continue the registration process I get the Error message "Please fill in this field". I am testing in the Incognito Browser of Brave.

**Comment by sxntaxis** [comment]
2025, i'm still wondering if there's a way to not show the email step on new registrations.
